File: app/src/main/java/com/platdmit/simplecloudmanager/domain/SCMApp.java
package com.platdmit.simplecloudmanager.domain;

import android.app.Application;

import com.platdmit.simplecloudmanager.data.api.implement.ApiAccountRepoImp;
import com.platdmit.simplecloudmanager.data.database.DbManager;
import com.platdmit.simplecloudmanager.domain.helpers.ActualApiKeyModule;
import com.platdmit.simplecloudmanager.domain.helpers.ActualApiKeyService;

import net.danlew.android.joda.JodaTimeAndroid;

import androidx.room.Room;

public class SCMApp extends Application {
    private static ActualApiKeyService mActualApiKeyService;
    private static DbManager mDb;

    @Override
    public void onCreate() {
        super.onCreate();

        //Init time library
        JodaTimeAndroid.init(this);

        //Init Db
        mDb = Room.databaseBuilder(this, DbManager.class, "sc_manager").build();

        //Init apiKey updater
        mActualApiKeyService = new ActualApiKeyModule(
                new ApiAccountRepoImp()
        );
    }

    public static ActualApiKeyService getActualApiKeyService(){
        return mActualApiKeyService;
    }

    public static DbManager getDb() {
        return mDb;
    }

}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/SCMApp.kt
package com.platdmit.simplecloudmanager.domain

import android.app.Application
import androidx.room.Room
import com.platdmit.simplecloudmanager.data.api.implement.ApiAccountRepoImp
import com.platdmit.simplecloudmanager.data.database.DbManager
import com.platdmit.simplecloudmanager.domain.helpers.ActualApiKeyModule
import com.platdmit.simplecloudmanager.domain.helpers.ActualApiKeyService
import net.danlew.android.joda.JodaTimeAndroid

class SCMApp : Application() {

    override fun onCreate() {
        super.onCreate()

        //Init time library
        JodaTimeAndroid.init(this)

        //Init Db
        db = Room.databaseBuilder(this, DbManager::class.java, "sc_manager").build()

        //Init apiKey updater
        actualApiKeyService = ActualApiKeyModule(
                ApiAccountRepoImp()
        )
    }

    companion object {

        lateinit var actualApiKeyService: ActualApiKeyService
            private set
        lateinit var db: DbManager
            private set

    }
}